source/simulator/helper_functions.cc

```diff
@@ -2311,10 +2340,22 @@ namespace aspect
                 fe_face_values.reinit (cell, face_number);
                 fe_face_values[introspection.extractors.velocities].get_function_values(current_linearization_point,
                                                                                         face_current_velocity_values);
-
                 // ... check if the face is an outflow boundary by integrating the normal velocities
                 // (flux through the boundary) as: int u*n ds = Sum_q u(x_q)*n(x_q) JxW(x_q)...
-                double integrated_flow = 0;
+                // do this for the solid velocity, the darcy velocity, and/or the fluid velocity.
+                double integrated_solid_flow = 0;
+                double integrated_darcy_flow = std::numeric_limits<double>::max();
+                double integrated_fluid_flow = std::numeric_limits<double>::max();
+
+                if (consider_darcy_velocity)
+                  {
+                    integrated_darcy_flow = 0;
+                    in.reinit(fe_face_values, cell, introspection, solution);
+                    material_model->evaluate(in, out);
+                  }
+
+                if (parameters.include_melt_transport)
+                  integrated_fluid_flow = 0;
```

Comment:
these two lines seem redundant.

Reply:
I do this because by default (no melt transport) `integrated_fluid_flow = std::numeric_limits<double>::max()` so that the if statement at line 2400 still works in solid only models. If there is melt though, then I need to set `integrated_fluid_flow = 0` otherwise line 2388:
```
integrated_fluid_flow += (face_current_fluid_velocity_values[q] * fe_face_values.normal_vector(q)) *
                                          fe_face_values.JxW(q);
```
will always be infinity. This was the only way I could think to get around this issue, but if you have a better approach that would be great!

Reply:
The same is done for when a field is advected using the darcy advection method at line 2352

Reply:
Oh, I see. I would put that into the initialization directly then.

Reply:
Ah yes! That will make this look much cleaner